zebra-state/src/service/finalized_state/zebra_db/transparent.rs

```diff
@@ -105,16 +105,83 @@ impl ZebraDb {
 
         self.db.zs_get(&utxo_by_out_loc, &output_location)
     }
+
+    /// Returns the unspent transparent outputs for a [`transparent::Address`],
+    /// if they are in the finalized state.
+    #[allow(dead_code)]
+    pub fn address_utxos(
+        &self,
+        address: &transparent::Address,
+    ) -> BTreeMap<OutputLocation, transparent::Utxo> {
+        let address_location = match self.address_location(address) {
+            Some(address_location) => address_location,
+            None => return BTreeMap::new(),
+        };
+
+        let output_locations = self.address_utxo_locations(address_location);
+
+        // Ignore any outputs spent by blocks committed during this query
+        output_locations
+            .iter()
+            .flat_map(|&addr_out_loc| {
+                Some((
+                    addr_out_loc.unspent_output_location(),
+                    self.utxo_by_location(addr_out_loc.unspent_output_location())?,
+                ))
+            })
+            .collect()
+    }
+
+    /// Returns the unspent transparent output locations for a [`transparent::Address`],
+    /// if they are in the finalized state.
+    pub fn address_utxo_locations(
+        &self,
+        address_location: AddressLocation,
+    ) -> BTreeSet<AddressUnspentOutput> {
+        let utxo_by_transparent_addr_loc =
+            self.db.cf_handle("utxo_by_transparent_addr_loc").unwrap();
+
+        // Manually fetch the entire addresses' UTXO locations
+        let mut addr_unspent_outputs = BTreeSet::new();
+
+        // An invalid key representing the minimum possible output
+        let mut unspent_output = AddressUnspentOutput::address_iterator_start(address_location);
+
+        loop {
+            // A valid key representing an entry for this address or the next
+            unspent_output = match self
+                .db
+                .zs_next_key_value_from(&utxo_by_transparent_addr_loc, &unspent_output)
+            {
+                Some((unspent_output, ())) => unspent_output,
+                // We're finished with the final address in the column family
+                None => break,
+            };
+
+            // We found the next address, so we're finished with this address
+            if unspent_output.address_location() != address_location {
+                break;
+            }
+
+            addr_unspent_outputs.insert(unspent_output);
+
+            // A potentially invalid key representing the next possible output
+            unspent_output.address_iterator_next();
+        }
```

Comment:
It's possible, but I'd rather fix the locking bug with the RPCs, then use [`rocksdb::DBIterator`](https://docs.rs/rocksdb/latest/rocksdb/struct.DBIteratorWithThreadMode.html) directly.

I was waiting until we had all our database changes done before looking at the bug. Because it might not be a high priority, or we might have accidentally fixed it.